ui/app/services/control-group.js

```diff
@@ -0,0 +1,56 @@
+import Ember from 'ember';
+import ControlGroupError from 'vault/lib/control-group-error';
+
+const { Service, assign, inject, RSVP } = Ember;
+
+// list of endpoints that return wrapped responses
+// without `wrap-ttl`
+const WRAPPED_RESPONSE_PATHS = [
+  'sys/wrapping/rewrap',
+  'sys/wrapping/wrap',
+  'sys/replication/performance/primary/secondary-token',
+  'sys/replication/dr/primary/secondary-token',
+];
+
+export default Service.extend({
+  version: inject.service(),
+  router: inject.service(),
+
+  checkForControlGroup(callbackArgs, response, wasWrapTTLRequested) {
+    let creationPath = response && Ember.get(response, 'wrap_info.creation_path');
+    if (
+      this.get('version.isOSS') ||
+      wasWrapTTLRequested ||
+      !response ||
+      (creationPath && WRAPPED_RESPONSE_PATHS.includes(creationPath)) ||
+      (response && !response.wrap_info)
+    ) {
+      return RSVP.resolve(...callbackArgs);
+    }
+    let error = new ControlGroupError();
+    error = assign(error, response.wrap_info);
```

Comment:
It's great that you have a custom error class. But is it necessary to flatten all attributes from `wrap_info` into the error? It might be clearer to be more explicit here and in the constructor. I had to dig around the tests to figure out what the expected attributes on `wrap_info` were.